sql/updates/world/3.3.5/2016_04_18_09_world_335.sql

```diff
@@ -0,0 +1,34 @@
+-- Correct gossip_menu_options for NPC 3836 Mountaineer Pebblebitty / Quest 3181
+DELETE FROM `gossip_menu_option` WHERE `menu_id` IN (1201,1202,1203,1204,1205,1206) AND `id` = 0;
+INSERT INTO `gossip_menu_option` (`menu_id`,`id`,`option_icon`,`option_text`,`OptionBroadcastTextID`,`option_id`,`npc_option_npcflag`,`action_menu_id`,`action_poi_id`,`box_coded`,`box_money`,`box_text`,`BoxBroadcastTextID`) VALUES
+(1201, 0,0, "What's the worst that could happen?", 4266, 1,1, 1202, 0,0,0,'',0),
+(1202, 0,0, "Another way? Do tell...",             4268, 1,1, 1203, 0,0,0,'',0),
+(1203, 0,0, "Orcs? Badlands? I'm invulnerable!",   4270, 1,1, 1204, 0,0,0,'',0),
+(1204, 0,0, "Absolutely!",                         4272, 1,1, 1205, 0,0,0,'',0),
+(1205, 0,0, "My apologies, Pebblebitty.",          4274, 1,1, 1206, 0,0,0,'',0),
+(1206, 0,0, "Done, done, and done.",               4276, 1,1,    0, 0,0,0,'',0);
+
+-- link gossip_menu_option menu_id to npc_text ID via gossip_menu (1201,1833 already exists)
+DELETE FROM `gossip_menu` WHERE `entry` IN (1202,1203,1204,1205,1206);
+INSERT INTO `gossip_menu` (`entry`,`text_id`) VALUES
+(1202,1834),
+(1203,1835),
+(1204,1836),
+(1205,1837),
+(1206,1838);
+
+-- SmartAI script for creature entry 3836 Mountaineer Pebblebitty
+UPDATE `creature_template` SET `AIName`= 'SmartAI',`ScriptName`= '' WHERE `entry`= 3836;
+DELETE FROM `smart_scripts` WHERE `source_type` = 0 AND  `entryorguid` = 3836;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(3836,0, 0,0, 62,0,100,0,1201,0,0,0, 98, 1202, 1834,0,0,0,0, 7,0,0,0,0,0,0,0,'Mountaineer Pebblebitty - on gossip option 1201 selected - send gossip 1834'),
+(3836,0, 1,0, 62,0,100,0,1202,0,0,0, 98, 1203, 1835,0,0,0,0, 7,0,0,0,0,0,0,0,'Mountaineer Pebblebitty - on gossip option 1202 selected - send gossip 1835'),
+(3836,0, 2,0, 62,0,100,0,1203,0,0,0, 98, 1204, 1836,0,0,0,0, 7,0,0,0,0,0,0,0,'Mountaineer Pebblebitty - on gossip option 1203 selected - send gossip 1836'),
+(3836,0, 3,0, 62,0,100,0,1204,0,0,0, 98, 1205, 1837,0,0,0,0, 7,0,0,0,0,0,0,0,'Mountaineer Pebblebitty - on gossip option 1204 selected - send gossip 1837'),
+(3836,0, 4,0, 62,0,100,0,1205,0,0,0, 98, 1206, 1838,0,0,0,0, 7,0,0,0,0,0,0,0,'Mountaineer Pebblebitty - on gossip option 1205 selected - send gossip 1838'),
+(3836,0, 5,0, 62,0,100,0,1206,0,0,0, 98, 1201, 1833,0,0,0,0, 7,0,0,0,0,0,0,0,'Mountaineer Pebblebitty - on gossip option 1206 selected - send gossip 1833');
```

Comment:
I know, that is why I wrote that I'm not satisfied with the current SQL. I haven't decided yet how to handle the difference between before and after going through the gossip dialogue. Correct me if I am wrong, but wouldn't it be better if only the `gossip_menu_option` is available when returning from Desolace with the quest active, so the player has to go through the gossip options one time first, making the "deliver quest" icon and gossip link available only after the dialogue is completed? The SAI needs at least one more line to make this happen, of course. I have not checked if this can be done yet, but I think this is what the quest needs to contain the correct logic. Maybe I need to write an additional condition line as well, I haven't tested yet.